splinterd/src/logging/mod.rs

```diff
@@ -0,0 +1,51 @@
+use log::Level;
+use serde::Deserialize;
+
+mod log4rs;
+
+#[derive(Deserialize, Clone, Debug)]
+pub struct LogConfig {
+    pub root: RootConfig,
+    pub appenders: Vec<AppenderConfig>,
+    pub loggers: Vec<LoggerConfig>,
+}
+
+#[derive(Deserialize, Clone, Debug)]
+pub struct LoggerConfig {
+    name: String,
+    appenders: Vec<String>,
+    level: Level,
+}
+
+#[derive(Deserialize, Clone, Debug)]
+pub struct RootConfig {
+    pub appenders: Vec<String>,
+    pub level: Level,
+}
+
+#[derive(Deserialize, Clone, Debug)]
+pub struct AppenderConfig {
+    pub name: String,
+    #[serde(default = "default_pattern")]
+    pub encoder: String,
+    pub target: LogTarget,
+}
+
+#[derive(Deserialize, Clone, Debug)]
+pub enum LogTarget {
+    StdOut,
+    StdErr,
+    File(String),
+    RollingFile(String),
+}
+
+#[derive(Deserialize, Clone, Debug)]
+pub struct LogConfigPartial {
+    root: Option<RootConfig>,
+    appenders: Option<Vec<AppenderConfig>>,
+    loggers: Option<Vec<LoggerConfig>>,
+}
+
+pub fn default_pattern() -> String {
```

Comment:
This should at most be a const declaration.

Reply:
It needs to be a function for its use within serde, so I have made the function private and added a public const with the actual pattern for use outside of this file.